runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -484,7 +485,11 @@ public void clear() {
     @Override
     @Accessor
     public @Nonnull Set<Entry<K, V>> entrySet() {
-        return ImmutableSet.copyOf(mainMap.entrySet());
+        Set<Entry<K, V>> entries = new HashSet<>();
+        for (Map.Entry<K, V> entry : mainMap.entrySet()) {
```

Comment:
Can you add some comments about what is going on here? (Trying to keep shallow references but not copying the map)

Reply:
Done.